bpf/kmesh/workload/sockops_tuple.c

```diff
@@ -134,59 +139,106 @@ static inline void enable_encoding_metadata(struct bpf_sock_ops *skops)
     if (err && err != -ENOENT)
-        BPF_LOG(ERR, KMESH, "record netcookie failed!, err is %d\n", err);
+        BPF_LOG(ERR, KMESH, "remove ip failed!, err is %d\n", err);
 }
 
-static inline bool skops_conn_from_cni_sim_add(struct bpf_sock_ops *skops)
+static inline bool conn_from_sim(struct bpf_sock_ops *skops, __u32 ip, __u32 port)
 {
-    // cni sim connect 0.0.0.1:929(0x3a1)
-    // 0x3a1 is the specific port handled by the cni for enable Kmesh
+    __u32 rev_port = bpf_ntohl(skops->remote_port);
+    __u32 client_ip = bpf_ntohl(skops->remote_ip4);
 #if !OE_23_03
-    return ((bpf_ntohl(skops->remote_ip4) == 1) && (bpf_ntohl(skops->remote_port) == 0x3a1));
-#else
-    return ((bpf_ntohl(skops->remote_ip4) == 1) && (bpf_ntohl(skops->remote_port) == 0x3a10000));
+    port >>= 16;
 #endif
+    return (client_ip == ip) && (port == rev_port);
+}
+
+static inline bool skops_conn_from_cni_sim_add(struct bpf_sock_ops *skops)
+{
+    // cni sim connect 0.0.0.1:929(0x3a1)
+    // 0x3a1 is the specific port handled by the cni to enable Kmesh
+    return conn_from_sim(skops, 1, ENABLE_KMESH_PORT);
 }
 
 static inline bool skops_conn_from_cni_sim_delete(struct bpf_sock_ops *skops)
 {
     // cni sim connect 0.0.0.1:930(0x3a2)
-    // 0x3a2 is the specific port handled by the cni for disable Kmesh
-#if !OE_23_03
-    return ((bpf_ntohl(skops->remote_ip4) == 1) && (bpf_ntohl(skops->remote_port) == 0x3a2));
-#else
-    return ((bpf_ntohl(skops->remote_ip4) == 1) && (bpf_ntohl(skops->remote_port) == 0x3a20000));
-#endif
+    // 0x3a2 is the specific port handled by the cni to disable Kmesh
+    return conn_from_sim(skops, 1, DISABLE_KMESH_PORT);
+}
+
+static inline bool skops_conn_from_bypass_sim_add(struct bpf_sock_ops *skops)
+{
+    // bypass sim connect 0.0.0.1:931(0x3a3)
+    // 0x3a3 is the specific port handled by daemon to enable bypass
+    return conn_from_sim(skops, 1, ENABLE_BYPASS_PORT);
+}
+
+static inline bool skops_conn_from_bypass_sim_delete(struct bpf_sock_ops *skops)
+{
+    // bypass sim connect 0.0.0.1:932(0x3a4)
+    // 0x3a4 is the specific port handled by the daemon to disable bypass
+    return conn_from_sim(skops, 1, DISABLE_BYPASS_PORT);
+}
+
+static inline void set_bypass_value(__u32 ip, int new_bypass_value)
+{
+    __u64 key = ip;
+    manager_value_t *current_value = bpf_map_lookup_elem(&map_of_manager, &key);
+    if (!current_value || current_value->is_bypassed == new_bypass_value)
+        return;
+
+    current_value->is_bypassed = new_bypass_value;
+
+    int err = bpf_map_update_elem(&map_of_manager, &key, current_value, BPF_EXIST);
+    if (err)
+        BPF_LOG(ERR, KMESH, "set bypass value failed!, err is %d\n", err);
 }
 
 static inline bool ipv4_mapped_addr(__u32 ip6[4])
 {
     return ip6[0] == 0 && ip6[1] == 0 && ip6[2] == 0xFFFF0000;
 }
 
+static inline void skops_handle_kmesh_managed_process(struct bpf_sock_ops *skops)
+{
+    if (skops_conn_from_cni_sim_add(skops))
+        record_ip(skops->local_ip4);
```

Comment:
can we also rename `record_ip`, we cannot get the meaning of it

Reply:
> can we also rename `record_ip`, we cannot get the meaning of it

done